net-p2p/retroshare/retroshare-0.6.9999.ebuild

```diff
@@ -14,131 +14,101 @@ LICENSE="GPL-2 GPL-3 Apache-2.0 LGPL-2.1"
 SLOT="0"
 KEYWORDS=""
 
-IUSE="cli feedreader gnome-keyring +gui voip"
+IUSE="cli control-socket gnome-keyring +gui settings-api +sqlcipher +webui"
 REQUIRED_USE="
 	|| ( cli gui )
-	feedreader? ( gui )
-	voip? ( gui )"
+	settings-api? ( || ( control-socket webui ) )"
 
 RDEPEND="
 	app-arch/bzip2
-	dev-db/sqlcipher
-	dev-libs/openssl:0
-	dev-qt/qtcore:5
-	dev-qt/qtmultimedia:5
-	dev-qt/qtnetwork:5
-	dev-qt/qtprintsupport:5
-	dev-qt/qtscript:5
-	dev-qt/qtxml:5
-	net-libs/libmicrohttpd
-	net-libs/libupnp:0
+	dev-libs/openssl:0=
+	>=dev-libs/rapidjson-1.1.0
+	net-libs/libupnp:0=
 	sys-libs/zlib
+	control-socket? ( dev-qt/qtnetwork:5 )
 	gnome-keyring? ( gnome-base/libgnome-keyring )
-	feedreader? (
-		dev-libs/libxml2
-		dev-libs/libxslt
-		net-misc/curl
-	)
 	gui? (
-		dev-qt/designer:5
+		dev-qt/qtcore:5
+		dev-qt/qtmultimedia:5
+		dev-qt/qtnetwork:5
+		dev-qt/qtprintsupport:5
+		dev-qt/qtscript:5
+		dev-qt/qtxml:5
 		dev-qt/qtgui:5
 		dev-qt/qtwidgets:5
 		dev-qt/qtx11extras:5
 		x11-libs/libX11
 		x11-libs/libXScrnSaver
 	)
-	voip? (
-		media-libs/opencv[-qt4(-)]
-		media-libs/speex
-		virtual/ffmpeg[encode]
-	)"
-DEPEND="${RDEPEND}
-	virtual/pkgconfig"
-
-src_prepare() {
-	local dir
-
-	sed -i \
-		-e "s|/usr/lib/retroshare/extensions6/|/usr/$(get_libdir)/${PN}/extensions6/|" \
-		libretroshare/src/rsserver/rsinit.cc \
-		|| die "sed on libretroshare/src/rsserver/rsinit.cc failed"
+	settings-api? ( dev-qt/qtcore:5 )
+	sqlcipher? ( dev-db/sqlcipher )
+	!sqlcipher? ( dev-db/sqlite )
+	webui? ( net-libs/libmicrohttpd )"
 
-	rs_src_dirs="libbitdht/src openpgpsdk/src libresapi/src libretroshare/src supportlibs/pegmarkdown"
-	use cli && rs_src_dirs="${rs_src_dirs} retroshare-nogui/src"
-	use feedreader && rs_src_dirs="${rs_src_dirs} plugins/FeedReader"
-	use gui && rs_src_dirs="${rs_src_dirs} retroshare-gui/src"
-	use voip && rs_src_dirs="${rs_src_dirs} plugins/VOIP"
-
-	# Force linking to sqlcipher ONLY
-	sed -i \
-		-e '/isEmpty(SQLCIPHER_OK) {/aerror(libsqlcipher not found)' \
-		retroshare-gui/src/retroshare-gui.pro \
-		retroshare-nogui/src/retroshare-nogui.pro || die 'sed on retroshare-gui/src/retroshare-gui.pro failed'
-
-	# Avoid openpgpsdk false dependency on qtgui
-	sed -i '2iQT -= gui' openpgpsdk/src/openpgpsdk.pro || die
-
-	eapply_user
-}
+DEPEND="${RDEPEND}
+	gui? ( dev-qt/designer:5 )
+	dev-qt/qtcore:5
+	virtual/pkgconfig
+"
 
 src_configure() {
-	for dir in ${rs_src_dirs} ; do
-		pushd "${S}/${dir}" >/dev/null || die
-		eqmake5 $(use gnome-keyring && echo CONFIG+=rs_autologin)
-		popd >/dev/null || die
-	done
-}
+	local qConfigs=()
 
-src_compile() {
-	local dir
-
-	for dir in ${rs_src_dirs} ; do
-		emake -C "${dir}"
-	done
+	use cli && qConfigs+="retroshare_nogui" || qConfigs+="no_retroshare_nogui"
```

Comment:
Please use `foo+=( ... )` to append to an array. I'm surprised this works at all.

That said, for yes/no cases you can do something like:

    qConfigs+=( $(usex ... '' 'no_')foobarbaz )